admiral/pkg/clusters/serviceentry.go

```diff
@@ -306,46 +321,85 @@ func AddServiceEntriesWithDr(cache *AdmiralCache, sourceClusters map[string]stri
+					deleteDestinationRule(oldDestinationRule, syncNamespace, rc)
+				} else {
 
-			// if no endpoint, delete this SE in all dependency clusters and remove the cash in map serviceEntries
-			if len(se.Endpoints) == 0 {
-				deleteServiceEntry(oldServiceEntry, syncNamespace, rc)
-				// after deleting the service entry, destination rule also need to be deleted if the service entry host no longer exists
-				deleteDestinationRule(oldDestinationRule, syncNamespace, rc)
-			} else {
-				newServiceEntry := createServiceEntrySkeletion(*se, serviceEntryName, syncNamespace)
-
-				//Add a label
-				var identityId string
-				if identityValue, ok := cache.CnameIdentityCache.Load(se.Hosts[0]); ok {
-					identityId = fmt.Sprint(identityValue)
-					newServiceEntry.Labels = map[string]string{common.GetWorkloadIdentifier(): fmt.Sprintf("%v", identityId)}
-				}
+					newServiceEntry := createServiceEntrySkeletion(*seDr.ServiceEntry, seDr.SeName, syncNamespace)
 
-				addUpdateServiceEntry(newServiceEntry, oldServiceEntry, syncNamespace, rc)
+					if newServiceEntry != nil {
+						newServiceEntry.Labels = map[string]string{common.GetWorkloadIdentifier(): fmt.Sprintf("%v", identityId)}
+						addUpdateServiceEntry(newServiceEntry, oldServiceEntry, syncNamespace, rc)
+					}
 
-				// if event was deletion when this function was called, then GlobalTrafficCache should already deleted the cache globalTrafficPolicy is an empty shell object
-				globalTrafficPolicy := cache.GlobalTrafficCache.GetFromIdentity(identityId, strings.Split(se.Hosts[0], ".")[0])
-				destinationRule := getDestinationRule(se.Hosts[0], rc.NodeController.Locality.Region, globalTrafficPolicy)
-				newDestinationRule := createDestinationRuleSkeletion(*destinationRule, destinationRuleName, syncNamespace)
+					newDestinationRule := createDestinationRuleSkeletion(*seDr.DestinationRule, seDr.DrName, syncNamespace)
+					// if event was deletion when this function was called, then GlobalTrafficCache should already deleted the cache globalTrafficPolicy is an empty shell object
+					addUpdateDestinationRule(newDestinationRule, oldDestinationRule, syncNamespace, rc)
+				}
+			}
+		}
+	}
+}
 
-				addUpdateDestinationRule(newDestinationRule, oldDestinationRule, syncNamespace, rc)
+func createSeAndDrSetFromGtp(env, region string, se *networking.ServiceEntry, globalTrafficPolicy *v1.GlobalTrafficPolicy,
+	cache *AdmiralCache) map[string]*SeDrTuple {
+	var defaultDrName = se.Hosts[0] + "-default-dr"
+	var defaultSeName = se.Hosts[0] + "-se"
+	var seDrSet = make(map[string]*SeDrTuple)
+	if globalTrafficPolicy != nil {
+		gtp := globalTrafficPolicy.Spec
+		for _, gtpTrafficPolicy := range gtp.Policy {
+			var modifiedSe *networking.ServiceEntry
+			var host = se.Hosts[0]
+			var drName, seName = defaultDrName, defaultSeName
+			if gtpTrafficPolicy.Dns != env && gtpTrafficPolicy.Dns != common.Default {
```

Comment:
Do you possible mean some other comparison here? Is it possible that the DNS = env?